release/Jenkinsfile

```diff
@@ -12,63 +15,70 @@ node {
-
-                        sh 'mvn -T 1C  clean install -Djacoco.skip=true -DskipTests -Dfinbugs.skip=true'
-                    }
-
-                    //Prepare for release, release on github (tag)
-                    stage('mvn release:prepare') {
-                        writeSettingsXmlFile()
-                        sh 'mvn -s settings.xml release:prepare -B -Darguments="-DskipTests" -DautoVersionSubmodules=true -Djacoco.skip=true -Dfinbugs.skip=true -Dusername=$GITHUB_USERNAME -Dpassword=$GITHUB_TOKEN'
-                    }
-
-                    //Release to maven repo
-                    stage('mvn release:perform') {
-                        sh 'mvn -s settings.xml release:perform -Djacoco.skip=true'
-                    }
-                } catch (err) {
-                    currentBuild.result = "FAILURE"
-
-                    if (err.toString().contains('AbortException')) {
-                        currentBuild.result = "ABORTED"
-                    }
-
-                } finally {
-                    deleteDir()
-                    //TODO: send notification to gitter/hipchat etc.
+
+    def gitNameEmail = (env.GIT_AUTHOR_AND_GIT_EMAIL).split(':')
+    println gitNameEmail[0] + ":" + gitNameEmail[1]
+    String gitName = gitNameEmail[0]
+    String gitEmail = gitNameEmail[1]
+    withEnv(["GIT_COMMITTER_EMAIL=$gitEmail", "GIT_COMMITTER_NAME=$gitName", "GIT_AUTHOR_NAME=$gitName", "GIT_AUTHOR_EMAIL=$gitEmail"]) {
+        withCredentials([
+                usernamePassword(credentialsId: '${JENKINS_CREDENTIALS}', passwordVariable: 'JENKINS_PASSWORD', usernameVariable: 'JENKINS_USERNAME'),
+                usernamePassword(credentialsId: '${GITHUB_CREDENTIALS}', passwordVariable: 'GITHUB_TOKEN', usernameVariable: 'GITHUB_USERNAME'),
+        ]) {
+            try {
+                // checkout from git
+                stage('Checkout') {
+
+                    echo "deploying as $JENKINS_USERNAME"
+                    git url: '${GIT_REPO_URL}', branch: '${BRANCH}'
+
+                    // setup git creds
+                    sh 'echo https://$GITHUB_USERNAME:$GITHUB_TOKEN@github.com > .git-credentials'
+                    sh 'cat .git-credentials'
+                    sh "git config credential.helper 'store --file .git-credentials'"
+
+                    //Make sure local branch is tracking remote. This enables releasing off branch other than master
+                    sh 'git push -u origin $BRANCH'
+                }
+
+                stage('Clean') {
+                    sh 'mvn clean -DcleanNode'
+                }
+
+                //Sanity build
+                stage('Build') {
+                    sh 'echo Java version: `java -version`'
+                    sh 'echo Maven version: `mvn --version`'
+                    sh 'echo Git version: `git --version`'
+
+                    // clean build
+                    sh 'mvn -B -T 1C clean install -Djacoco.skip=true -DskipTests -Dfindbugs.skip=true'
```

Comment:
@michaelneale done. Also removed `mvn clean` hack which was apparently due to maven frontend end concurrency issue. PTAL.